test/helpers/cliCommandRunner.js

```diff
@@ -0,0 +1,128 @@
+// Import Node.js Dependencies
+import { fork } from "node:child_process";
+import { createInterface } from "node:readline";
+
+// Import Third-party Dependencies
+import { MockAgent, setGlobalDispatcher } from "undici";
+
+// Import Internal Dependencies
+import { getCurrentRepository } from "../../src/commands/scorecard.js";
+
+function initChildProcess({ path, args, mockApiOptions }) {
+  const childProcess = fork(path, args, {
+    stdio: ["ignore", "pipe", "pipe", "ipc"]
+  });
+  childProcess.send(mockApiOptions);
+
+  return childProcess;
+}
+
+async function forkAndGetLines(options) {
+  const childProcess = initChildProcess(options);
+  const rStream = createInterface(childProcess.stdout);
+  const lines = [];
+
+  for await (const line of rStream) {
+    lines.push(line);
+  }
+
+  if (!options.keepAlive) {
+    childProcess.kill();
+  }
+
+  return lines;
+}
+
+function getMockApiOptions(pkgName, options) {
+  // use faker.js
+  const defaultIntercept = {
+    path: `/projects/github.com/${pkgName}`,
+    method: "GET"
+  };
+
+  const defaultBody = {
+    date: "2222-12-31",
+    repo: {
+      name: `github.com/${pkgName}`
+    },
+    score: 5.2,
+    checks: [
+      {
+        name: "Maintained",
+        score: 10,
+        reason: "Package is maintained"
+      }
+    ]
+  };
+
+  const { response, intercept: customIntercept, baseUrl } = options;
+  const intercept = customIntercept || defaultIntercept;
+  const body = response?.body ?? defaultBody;
+  const status = response?.status ?? 200;
+
+  return {
+    baseUrl,
+    intercept,
+    response: {
+      status,
+      body
+    }
+  };
+}
+
+export async function runCliCommand(cb, args = []) {
+  process.on("message", async(mockApi) => {
+    if (!mockApi) {
+      await cb(...args);
+
+      process.exit(0);
+    }
+
+    const kScorecardAgent = new MockAgent();
+    const kScorecardPool = kScorecardAgent.get(mockApi.baseUrl);
+    kScorecardAgent.disableNetConnect();
+
+    const request = mockApi.intercept;
+    const response = mockApi.response;
```

Comment:
Is that really required? Feel like assigned variable just make thing harder to read.